pkg/controller/seed-controller-manager/constraint-controller/controller.go

```diff
@@ -216,15 +230,23 @@ func (r *reconciler) patchFinalizer(ctx context.Context, constraint *kubermaticv
 
 func (r *reconciler) reconcile(ctx context.Context, constraint *kubermaticv1.Constraint, log *zap.SugaredLogger) error {
 
+	clusterList, unwantedClusterList, err := r.getClustersForConstraint(ctx, constraint)
```

Comment:
Listing all clusters is something that is needed both for CE and EE editions, I would factor it out. I would do something like:

```
	clusterList := &kubermaticv1.ClusterList{}
	if err := r.seedClient.List(ctx, clusterList, &ctrlruntimeclient.ListOptions{LabelSelector: r.workerNameLabelSelector}); err != nil {
		return nil, nil, fmt.Errorf("failed listing clusters: %w", err)
	}
	wanted, unwanted := r.filterClusters(clusterList, constraint)
```

WDYT?